cobigen-cli/src/main/java/com/devonfw/cobigen/cli/commands/UpdateCommand.java

```diff
@@ -0,0 +1,171 @@
+import com.devonfw.cobigen.impl.generator.InputInterpreterImpl;
+
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * This class handle update command
+ */
+@Command(description = MessagesConstants.GENERATE_DESCRIPTION, name = "update", aliases = { "u" },
+    mixinStandardHelpOptions = true)
+public class UpdateCommand implements Callable<Integer> {
+    InputInterpreterImpl inputInterpreter = new InputInterpreterImpl();
+
+    public UpdateCommand() {
+        super();
+    }
+
+    /**
+     * This option provide specified list of template
+     */
+    @Option(names = { "--templates", "-t" }, split = ",", description = MessagesConstants.TEMPLATES_OPTION_DESCRIPTION)
+    /**
+     * Initialize templates variable
+     */
+    ArrayList<String> templates = null;
+
+    /**
+     * If this options is enabled, we will print also debug messages
+     */
+    @Option(names = { "--verbose", "-v" }, negatable = true, description = MessagesConstants.VERBOSE_OPTION_DESCRIPTION)
+    boolean verbose;
+
+    /**
+     * This option provides the use of multiple available increments
+     */
+    @Option(names = { "update", "u" }, split = ",", description = MessagesConstants.UPDATE_OPTION_DESCRIPTION)
+    /**
+     * Utils class for CobiGen related operations
+     */
+    private CobiGenUtils cobigenUtils = new CobiGenUtils();
+
+    /**
+     * Logger to output useful information to the user
+     */
+    private static Logger logger = LoggerFactory.getLogger(CobiGenCLI.class);
+
+    @Override
+    public Integer call() throws Exception {
+        Model model = null;
+        List<String> versionProvider = new ArrayList<>();
+        List<String> needversionupdate = new ArrayList<>();
+        List<GenerableArtifact> userSelection = new ArrayList<>();
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        MavenXpp3Writer writer = new MavenXpp3Writer();
+        File locationCLI = new File(GenerateCommand.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+        Path rootCLIPath = locationCLI.getParentFile().toPath();
+        File pomFile = new CobiGenUtils().extractArtificialPom(rootCLIPath);
+        if (pomFile.exists()) {
+            model = reader.read(new FileReader(pomFile));
+            List<Dependency> localPomDepedency = model.getDependencies();
+            int count = 0;
+            logger.info("(0) " + "All");
+            for (Dependency lclDependency : localPomDepedency) {
+                String[] localVersion = lclDependency.getVersion().split("\\.");
+                if (!lclDependency.getArtifactId().equals("freemarker")
+                    && !lclDependency.getArtifactId().equals("mmm-code-java-parser")
+                    && !lclDependency.getArtifactId().equals("mmm-code-base")
+                    && !lclDependency.getArtifactId().equals("mmm-code-java-maven")
+                    && !lclDependency.getArtifactId().equals("mmm-util-core")) {
```

Comment:
Please create a method called `Boolean dependencyShouldBeUpdated(lclDependenct.getGroupId)` which returns true when the group id of the dependency equals `com.devonfw.cobigen`. Please take a look to [this method](https://github.com/devonfw/tools-cobigen/blob/master/cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenVersionProvider.java#L64), which follows the same logic.